extensions/apns/apns_message_handler.go

```
@@ -245,6 +246,29 @@ func (a *APNSMessageHandler) parseKafkaMessage(message interfaces.KafkaMessage)
 	return notification, nil
 }
 
+func (a *APNSMessageHandler) createDedupContentFromPayload(notification *pusherAPNSKafkaMessage) string {
+	if notification.Payload == nil {
+		return "empty-payload"
+	}
+
+	payloadMap, ok := notification.Payload.(map[string]interface{})
+	if !ok {
+		return "invalid-payload-type"
+	}
+
+	// Extract only the APS part (the actual notification content)
+	if aps, exists := payloadMap["aps"]; exists {
+		apsJSON, err := json.Marshal(aps)
+		if err != nil {
+			a.Logger.WithError(err).Error("error marshalling APS for deduplication")
+			return "error-marshalling-aps"
+		}
+		return string(apsJSON)
```

why don't we return `string, error`? Then you discard the error in the caller function but avoids this quirk thing of returning a hardcoded error as a string. If you return a string then it's guaranteed that there was no error, if string is empty than it's guaranteed that there was an error. Then you adjust what field you'll be storing to determine if notification is duplicated or not

---

indeed much better, thanks Pedrao
I guess I just wanted to see what string was coming up on the LogZ as the notification message once it got into dedup, ended up making this quirky mess